exhibitors/models.py

```diff
@@ -56,53 +66,73 @@ class Value(models.Model):
-def random_path(a,b):
-    return ""
-
-def format_png():
-    return None
 
 # Info about an exhibitor to be displayed in apps and on website
 class CatalogInfo(models.Model):
-    exhibitor = models.OneToOneField(Exhibitor, on_delete=models.CASCADE)
+    exhibitor = models.OneToOneField(
+            Exhibitor,
+            on_delete=models.CASCADE,
+            primary_key=True,
+            )
     display_name = models.CharField(max_length=64)
-    slug = models.SlugField(db_index=False)
+    slug = models.SlugField(db_index=False, blank=True)
     short_description = models.CharField(max_length=64)
     description = models.TextField()
-    employees_sweden = models.IntegerField()
-    employees_world = models.IntegerField()
-    countries = models.IntegerField()
+    employees_sweden = models.IntegerField(default=0)
+    employees_world = models.IntegerField(default=0)
+    countries = models.IntegerField(default=0)
     website_url = models.CharField(max_length=128, blank=True)
     facebook_url = models.CharField(max_length=128, blank=True)
     twitter_url = models.CharField(max_length=128, blank=True)
     linkedin_url = models.CharField(max_length=128, blank=True)
 
-    # Images
+    # Image fields
+    # Field with name ending with original serves as the original uploaded
+    # image and should be the only image uploaded,
+    # the others are auto generated
+    logo_original = models.ImageField(
+            upload_to=UploadToDirUUID('exhibitors', 'logo_original'),
+            blank=True,
+            )
     logo_small = models.ImageField(
-            upload_to=random_path('exhibitors', 'logo_small'), blank=True)
+            upload_to=UploadToDir('exhibitors', 'logo_small'), blank=True)
     logo = models.ImageField(
-            upload_to=random_path('exhibitors', 'logo'), blank=True)
+            upload_to=UploadToDir('exhibitors', 'logo'), blank=True)
+
+    ad_original = models.ImageField(
+            upload_to=UploadToDirUUID('exhibitors', 'ad_original'), blank=True)
     ad = models.ImageField(
-            upload_to=random_path('exhibitors', 'ad'), blank=True)
+            upload_to=UploadToDir('exhibitors', 'ad'), blank=True)
 
+    # ManyToMany relationships
     programs = models.ManyToManyField('people.Programme', blank=True)
     main_work_field = models.ForeignKey(
             WorkField, blank=True, null=True, related_name='+')
     work_fields = models.ManyToManyField(
             WorkField, blank=True, related_name='+')
     job_types = models.ManyToManyField(JobType, blank=True)
+    continents = models.ManyToManyField(Continent, blank=True)
+    values = models.ManyToManyField(Value, blank=True)
 
     def __str__(self):
         return self.display_name
 
+    # Override default save method to automatically generate associated images
+    # if the original has been changed
     def save(self, *args, **kwargs):
+        if self.pk is None:
```

Comment:
That's not true if self.pk is 0 (maybe Django guarantees that primary keys starts at 1).